src/3D/LandIslandInterface.h

```diff
@@ -0,0 +1,52 @@
+/*******************************************************************************
+ * Copyright (c) 2018-2023 openblack developers
+ *
+ * For a complete list of all authors, please refer to contributors.md
+ * Interested in contributing? Visit https://github.com/openblack/openblack
+ *
+ * openblack is licensed under the GNU General Public License version 3.
+ *******************************************************************************/
+
+#pragma once
+#include <filesystem>
+
+#include <LNDFile.h>
+#include <glm/mat4x4.hpp>
+
+using LNDCell = openblack::lnd::LNDCell;
+
+namespace openblack
+{
+class LandBlock;
+namespace graphics
+{
+class FrameBuffer;
+class Texture2D;
+} // namespace graphics
+class LandIslandInterface
+{
+public:
+	virtual void LoadFromFile(const std::filesystem::path& path) = 0;
+	[[nodiscard]] virtual float GetHeightAt(glm::vec2) const = 0;
+	[[nodiscard]] virtual const LandBlock* GetBlock(const glm::u8vec2& coordinates) const = 0;
+	[[nodiscard]] virtual const LNDCell& GetCell(const glm::u16vec2& coordinates) const = 0;
+
+	// Debug
+	virtual void DumpTextures() const = 0;
+	virtual void DumpMaps() const = 0;
+
+	[[nodiscard]] virtual std::vector<LandBlock>& GetBlocks() = 0;
+	[[nodiscard]] virtual const std::vector<LandBlock>& GetBlocks() const = 0;
+	[[nodiscard]] virtual const std::vector<lnd::LNDCountry>& GetCountries() const = 0;
+
+	[[nodiscard]] virtual const graphics::Texture2D& GetAlbedoArray() const = 0;
+	[[nodiscard]] virtual const graphics::Texture2D& GetBump() const = 0;
+	[[nodiscard]] virtual const graphics::Texture2D& GetHeightMap() const = 0;
+	[[nodiscard]] virtual const graphics::FrameBuffer& GetFootprintFramebuffer() const = 0;
+
+	virtual void GetIndexExtent(glm::u16vec2& extentMin, glm::u16vec2& extentMax) const = 0;
+	virtual void GetOrthoViewProj(glm::mat4& view, glm::mat4& proj) const = 0;
+	virtual void GetExtent(glm::vec2& extentMin, glm::vec2& extentMax) const = 0;
```

Comment:
You don't have to do this in this PR but I don't like this pattern and would prefer an extent struct being returned.

Reply:
Hmm could you point me to what this could mean? I didn't find what the term extent struct means

Reply:
An extent is like a rectangle (in 2d, a rectangular cuboid in 3d) that is described by a minimum and maximum position.

```cpp
struct Extent2D
{
    glm::vec2 minima;
    glm::vec2 maxima;
}
```

Reply:
Wound't a tuple work better? Since the ExtentIndex returns glm::u16vec2& and GetOrthoViewProj returns glm::mat4
Or should i create a struct for each?
`struct Extent2D
	{
		glm::vec2 minimum;
		glm::vec2 maximum;
	};
    struct IndexExtent
	{
	    glm::u16vec2 minimum;
	    glm::u16vec2 maximum;
	};
	struct OrthoView
	{
	    glm::mat4 proj;
	    glm::mat4 view;
	};`
Like this?

Reply:
Problem with pairs and tuples is that members are unnamed and makes it hard to make sense of which data is which.

`GetOrthoViewProj` seems like it should be two getters.

Reply:
Fixed on next commit!
please take a look later